src/components/EditAccountModal/index.tsx

```diff
@@ -0,0 +1,189 @@
+import { Button, Submit } from "@components/Button";
+import * as yup from "yup";
+import { FormTextInput } from "@components/FormTextInput";
+import { SmallModalOverlay } from "@components/SmallModalOverlay";
+import { Controller, useForm } from "react-hook-form";
+import {
+  requiredUsernameValidation,
+  requiredEmailValidation,
+  requiredDisplaynameValidation,
+  optionalDescriptionValidation,
+  optionalLinkValidation,
+} from "@lib/formValidationUtil";
+import React, { FC, useEffect, useCallback } from "react";
+import { yupResolver } from "@hookform/resolvers/yup";
+import { FormTextarea } from "@components/FormTextarea";
+import { ParsedAccountType } from "@lib/hooks/usePublicAccounts";
+
+interface DefaultValuesType extends ParsedAccountType {
+  email: string;
+}
+
+export interface EditAccountModalPropType {
+  defaultValues: DefaultValuesType;
+  onSubmit?: (accountData: DefaultValuesType) => void;
+  submitButtonText?: string;
+  onCancel?: () => void;
+  cancelButtonText?: string;
+  onDelete?: () => void;
+  deleteButtonText?: string;
+}
+
+const formSchema = yup.object().shape({
+  username: requiredUsernameValidation,
+  displayName: requiredDisplaynameValidation,
+  email: requiredEmailValidation,
+  description: optionalDescriptionValidation,
+  link: optionalLinkValidation,
+});
+
+export const EditAccountModal: FC<EditAccountModalPropType> = ({
+  defaultValues,
+  onSubmit = () => undefined,
+  onCancel = () => undefined,
+  onDelete,
+  submitButtonText = "Speichern",
+  cancelButtonText = "Abbrechen",
+  deleteButtonText = "Löschen",
+}) => {
+  const {
+    control,
+    handleSubmit,
+    formState: { errors, dirtyFields },
```

Comment:
What are `dirtyFields`?

Reply:
Thank you for your review! Sorry for the late reply. Dirty fields are fields whose values have been changed and are unsaved. This terminology comes from the `useForm` hook from the [`react-hook-form`](https://react-hook-form.com/) package that we use to do form validation.